### PR TITLE
[CIS-1800] Make link preview tappable for URLs missing the URL scheme

### PR DESCRIPTION
**Pull request by bielikb**
### 🔗 Issue Links

CIS-1800

### 🎯 Goal

Opening URLs that missed `URL Scheme` would fail with error:
```
Failed to open URL [Google](http://google.com/) : Error Domain=NSOSStatusErrorDomain Code=-50 "invalid input parameters" UserInfo={NSDebugDescription=invalid input parameters, _LSLine=230, _LSFunction=-[_LSDOpenClient openURL:options:completionHandler:]}
```

The goal here was to leverage valid `URL` when user taps on the `image preview` of the link preview in the chat.

### 🛠 Implementation

* Adds new `url` computed property that prefers `titleLink` over `orgUrl`. Additionally it locally enriches the URL with the `http` scheme if it's missing.

### 🎨 Showcase

`google.com`
https://user-images.githubusercontent.com/4924709/174615584-793e726c-f9b5-42ac-aa85-794e0aef1925.mp4

### 🧪 Manual Testing Notes

**GIVEN** user opens a channel
**AND** the user sends link `google.com`

**WHEN** the user taps on the link
**THEN** user is navigated to the web browser
**WHEN** the user taps on the link preview
**THEN** the user is navigated to the web browser

### ☑️ Contributor Checklist
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [x] This change follows zero ⚠️ policy (required)
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS)

### 🎁 Meme

_Provide a funny gif or image that relates to your work on this pull request. (Optional)_

### CODE REVIEW

File: Tests/StreamChatTests/Extensions/URL_EnrichedURL_Tests.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        // GIVEN
+        let url = URL(string: "google.com")
+
+        // THEN
+        XCTAssertEqual(url?.enrichedURL.absoluteString, "http://google.com")
+    }
+
+    func test_urlWithSchemeIsNotChanged_whenHavingScheme() {
+        // GIVEN
+        let url = URL(string: "http://google.com")
```
Comment (evsaev) [inline Tests/StreamChatTests/Extensions/URL_EnrichedURL_Tests.swift:19]: Let's update a bit to ensure the existed scheme is preserved
```suggestion
        let url = URL(string: "https://google.com")
```

**Review by evsaev — APPROVED** [review]
🚢

File: Sources/StreamChat/Extensions/URL+EnrichedURL.swift [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import Foundation
+
+extension URL {
+    /// Enriches `URL` with `http` scheme if it's missing
+    var enrichedURL: URL {
+        guard scheme == nil else {
+            return self
+        }
+
+        return URL(string: "http://" + absoluteString) ?? self
```
Comment (adolfogarza) [inline Sources/StreamChat/Extensions/URL+EnrichedURL.swift:14]: Should the preferred URL Scheme be `https` for security reasons?

**Review by adolfogarza — APPROVED** [review]

File: Sources/StreamChat/Extensions/URL+EnrichedURL.swift (hunk 2 above)
Comment (bielikb) [reply Sources/StreamChat/Extensions/URL+EnrichedURL.swift:14]: This was explained on slack. Leveraging http scheme by default is safe.

**Pull request merged**